packages/p2p-v2/src/pages/my-ads/screens/MyAds/MyAdsTableRow/MyAdsTableRow.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { memo, useEffect, useState } from 'react';
+import clsx from 'clsx';
+import { useExchangeRateSubscription } from '@deriv/api';
+import { Button, Text, Tooltip } from '@deriv-com/ui';
+import { useDevice } from '../../../../../hooks';
+import { generateEffectiveRate } from '../../../../../utils/format-value';
+import DeactivateIcon from '../../../../../public/ic-archive.svg';
+import ActivateIcon from '../../../../../public/ic-unarchive.svg';
+import EditIcon from '../../../../../public/ic-edit.svg';
+import DeleteIcon from '../../../../../public/ic-delete.svg';
+import { PopoverDropdown } from '../../../../../components';
+import { ADVERT_TYPE, RATE_TYPE } from '../../../../../utils/constants';
+import { formatMoney } from '../../../../../utils/currency';
+import { AdStatus, AdType, ProgressIndicator } from '../../../components';
+import { TMyAdsTableRowRendererProps } from '../MyAdsTable/MyAdsTable';
+import './MyAdsTableRow.scss';
+
+const BASE_CURRENCY = 'USD';
+//TODO: to be modified after design is updated.
+const list = [
+    { label: 'Edit', value: 'edit' },
+    { label: 'Delete', value: 'delete' },
+    { label: 'Duplicate', value: 'duplicate' },
+    { label: 'Share', value: 'share' },
+    { label: 'Deactivate', value: 'deactivate' },
+];
+
+const MyAdsTableRow = ({ isBarred, isListed, onClickIcon, ...rest }: TMyAdsTableRowRendererProps) => {
+    const { isMobile } = useDevice();
+    const { data: exchangeRateValue, subscribe } = useExchangeRateSubscription();
+
+    const {
+        account_currency,
+        amount,
+        amount_display,
+        effective_rate,
+        id,
+        is_active,
+        local_currency,
+        max_order_amount_display,
+        min_order_amount_display,
+        payment_method_names,
+        price_display,
+        rate_display,
+        rate_type,
+        remaining_amount,
+        remaining_amount_display,
+        type,
+    } = rest;
+
+    useEffect(() => {
+        subscribe({
+            base_currency: BASE_CURRENCY,
+            target_currency: local_currency!,
+        });
+    }, [local_currency, subscribe]);
+
+    const [isActionsVisible, setIsActionsVisible] = useState(false);
+    const isAdvertListed = isListed && !isBarred;
+    const adPauseColor = isAdvertListed ? 'general' : 'less-prominent';
+    const amountDealt = (amount ?? 0) - (remaining_amount ?? 0);
+
+    const exchangeRate = exchangeRateValue?.rates?.[local_currency ?? ''];
+
+    const { displayEffectiveRate } = generateEffectiveRate({
+        price: Number(price_display),
+        rateType: rate_type,
+        rate: Number(rate_display),
+        localCurrency: local_currency,
+        exchangeRate,
+        marketRate: Number(effective_rate),
+    });
+
+    //TODO: get the floating rate configs after integration with usep2pconfig to handle disabled case.
```

Comment:
change it to integrating with useP2PSettings, cause we will be using this endpoint for any p2p_config related stuff